tests/Unit/NumericalAlgorithms/Interpolation/Test_InterpolateVariablesToPoints.cpp

```diff
@@ -0,0 +1,219 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/Interpolation/InterpolateVariablesToPoints.hpp"
+#include "NumericalAlgorithms/Spectral/LegendreGaussLobatto.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace TestTags {
+
+struct Scalar : db::DataBoxTag {
+  using type = ::Scalar<DataVector>;
+  static constexpr db::DataBoxString label = "Scalar";
+};
+
+template <size_t Dim>
+struct Vector : db::DataBoxTag {
+  using type = tnsr::I<DataVector, Dim, Frame::Grid>;
+  static constexpr db::DataBoxString label = "Vector";
+};
+
+}  // namespace TestTags
+
+namespace {
+
+void test_interpolate_to_points_1d(const Index<1>& mesh) noexcept {
+  using point = std::array<double, 1>;
+
+  // Test functions.  We fill a Variables with these functions,
```

Comment:
Look at Test_Divergence and see how I use a TensorProduct of PowX to easily setup these type of test functions.  Then you can use Tensor<DataVector> for the grid points of the mesh and the target points, and avoid loops over points, as well as getting straightforward approx checks